src/query/list_file.c

```diff
@@ -3862,9 +3890,19 @@ qfile_sort_list_with_func (THREAD_ENTRY * thread_p, QFILE_LIST_ID * list_id_p, S
 
   dup_option = ((option == Q_DISTINCT) ? SORT_ELIM_DUP : SORT_DUP);
 
+  /* find the query entry */
+  query_p = qmgr_get_query_entry (thread_p, list_id_p->query_id, LOG_FIND_THREAD_TRAN_INDEX (thread_p));
+
+  if (query_p == NULL)
+    {
+      /* query entry is not found */
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_QPROC_UNKNOWN_QUERYID, 1, list_id_p->query_id);
+      return NULL;
+    }
+
```

Comment:
Getting QMGR_QUERY_ENTRY in list_file may increase the complexity of modules. IMO, list file should not know the QMGR_QUERY_ENTRY structure.

Reply:
First,`QMGR_QUERY_ENTRY` has already accessed by `xqfile_get_list_file_page()` in `list_file.c`. What do you think about it? 
 Second, in our modifications, it is only accessed by `qfile_clear_list_id()` and `qfile_sort_list_with_func()`. In the formal, it is just for debugging purpose (assertion), and in the latter, I think it can be removed without much effort because the output qlist file has already TDE information.

Could you give your opinion? Do you think it should not be accessed at least in our modification?

Reply:
It makes sense. I'll chage it to use list files in the caller of `qmgr_get_query_entry()` like below:
`qfile_sort_list_with_func()`: `list_id_p`
`qexec_groupby()`: `output_list_id`
`qexec_execute_analytic()`: `output_list_id`
Could you review it if possible? There are some list files in each function and I wanna know your opinion about the the best list file to refer.

I expect all the list file is related to a query and have the information properly, but I'm not sure. So, I'll change it, do tests for each case, and notify you of the result.

Reply:
Thanks, commit: [40c862e](https://github.com/CUBRID/cubrid/pull/2450/commits/40c862e9dd9fbc679433bce67176530af7caab4a)